parsers/test/test_EIA.py

```diff
@@ -778,3 +781,36 @@ def test_texas_loses_one_mode(adapter, session):
     )
     data = EIA.fetch_production_mix(ZoneKey("US-TEX-ERCO"), session)
     assert len(data) == 0
+
+
+def test_fetch_returns_storage(adapter, session):
+    adapter.register_uri(
+        GET,
+        ANY,
+        json=json.loads(
+            resources.files("parsers.test.mocks.EIA")
+            .joinpath("US_CAL_IID-battery_storage.json")
+            .read_text()
+        ),
+    )
+
+    data_list = EIA.fetch_production_mix(ZoneKey("US-CAL-IID"), session)
+    expected = [
+        {
+            "zoneKey": "US-CAL-IID",
+            "source": "eia.gov",
+            "production": {
+                "gas": 1,
+                "coal": 1,
+                "wind": 1,
+                "hydro": 1,
+                "nuclear": 1,
+                "oil": 1,
+                "unknown": 1,
+                "solar": 1,
+                "geothermal": 1,
+            },
+            "storage": {"battery": 1, "hydro": 1},
```

Comment:
FYI this means it's charging, negative is dischargin. It's backwards, I know. But to change it we need some deeper re-structuring.

Reply:
makes sense, the parser is producing negative numbers. Should I change the test? From what I can tell what will happen if I set the fixture to have a negative value, then all the other production values will become negative aswell.

Reply:
No I think it's fine for now, but I think we will have to take a deeper look at this parser, possibly re-build it from the core and then we can add proper tests along the way.<br><br>I not confident enough in our test coverage here to just go about and change things in the future.

Reply:
Agree 100%, I'm writing a pitch for the next week that addresses this. If nothing else there's way too much processing logic happening at the request level here.

Reply:
Most of it should already be covered by our own parser classes as well or should be covered by them (like the all None values). So I think we can slim this down a lot!